### PR TITLE
Add t1-smartswitch topo support 

### PR DESCRIPTION
**Pull request by zjswhhh**
### Description of PR

Summary:
Fixes # (issue)

This PR is to add t1-smartswitch support.

1. add a t1-smartswich definition
2. add extra step `start_dpu_vm` to create sonic-vs neighbor to mimic DPUs
3. update `vm_topology` to bind DPUs interfaces to the bridges too 
4. update `annouce_route` modules to create exabgp processes on PTF for DPU neighbors too. Refer [testbed routing](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.testbed.Routing.md) for details. 
5. update `topo_facts` and `config_sonic_basedon_testbed` modules to update gen-mg logics

sign-off: Jing Zhang zhagjing@microsoft.com 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### How did you verify/test it?
Deployed a `t1-smartswitch` testbed on my dev host.

### CODE REVIEW

File: ansible/vtestbed.yaml [hunk 1]
```diff
@@ -336,3 +336,18 @@
   inv_name: veos_vtb
   auto_recover: False
   comment: Tests virtual switch vm as DPU
+
+- conf-name: vms-kvm-t1-dpu
```
Comment (r12f) [inline ansible/vtestbed.yaml:340]: rename t1-dpu to t1-smartswitch

Comment (zjswhhh) [reply ansible/vtestbed.yaml]: addressed

File: ansible/veos_vtb [hunk 2]
```diff
@@ -278,7 +281,7 @@ vm_host_1:
   hosts:
     STR-ACS-VSERV-01:
       ansible_host: 172.17.0.1
-      ansible_user: use_own_value
+      ansible_user: zhangjing
       vm_host_user: use_own_value
```
[the remark this answers is not in the document]
Comment (zjswhhh) [reply ansible/veos_vtb]: addressed

**Comment by ganglyu** [comment]
How to support other t1 topo like t1-8-lag, t1-28-lag, t1-32-lag?

File: ansible/vars/topo_t1-smartswitch.yml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        ipv6: fc00::46/126
+    bp_interface:
+      ipv4: 10.10.246.18/24
+      ipv6: fc0a::25/64
+
+  SONIC01DPU:
+    properties:
+    - common
+    - SonicHost
+    bgp:
```
Comment (ganglyu) [inline ansible/vars/topo_t1-smartswitch.yml:133]: Why do we have bgp config for DPU?

File: ansible/vars/topo_t1-smartswitch.yml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        - 10.0.0.38
+        - FC00::4D
+    interfaces:
+      Loopback0:
+        ipv4: 100.1.0.19/32
+        ipv6: 2064:100::13/128
+      Ethernet0:
+         ipv4: 10.0.0.37/31
+         ipv6: fc00::4a/126
+      Ethernet4:
```
Comment (ganglyu) [inline ansible/vars/topo_t1-smartswitch.yml:148]: I have checked lab dut, DPU only has 1 port connected with NPU.
Why does this DPU has two ports? Ethernet0 and Ethernet4

File: ansible/veos_vtb [hunk 5]
```diff
@@ -17,6 +17,7 @@ all:
           - t1-8-lag
           - t1-64-lag-clet
           - t1-backend
+          - t1-smartswitch
```
Comment (ganglyu) [inline ansible/veos_vtb:20]: How to support other t1 topo like t1-8-lag, t1-28-lag, t1-32-lag?

File: ansible/vars/topo_t1-smartswitch.yml (hunk 3 above)
Comment (zjswhhh) [reply ansible/vars/topo_t1-smartswitch.yml:133]: Topo file parser takes neighbor configuration from `bgp` config.

File: ansible/vars/topo_t1-smartswitch.yml (hunk 4 above)
Comment (zjswhhh) [reply ansible/vars/topo_t1-smartswitch.yml:148]: Agree that real DPU should only have one interface connected to NPU.

This one is an ad-hoc change to support BMv2 image.

File: ansible/veos_vtb (hunk 5 above)
Comment (zjswhhh) [reply ansible/veos_vtb:20]: Define port channel in topo files. 

When I raised this PR, we were focusing on vtestbed and VPP image. LAG isn't supported on sonic-vpp yet.

Comment (prabhataravind) [reply ansible/veos_vtb:20]: @zjswhhh could you consider renaming this topo accordingly to reflect that this is a vtestbed? We might need to reserve "t1-smartswitch" for the actual physical smartswitch testbeds.

**Comment by zjswhhh** [comment]
Hi @wangxin  -can you please help review?

File: ansible/veos_vtb (hunk 5 above)
Comment (prabhataravind) [reply ansible/veos_vtb:20]: Hi @zjswhhh, sure separate PR is fine. The reason I suggested this is there are some upstream commits that assume "t1-smartswitch" topo name for generating default config for physical smartswitch. 

For eg: https://github.com/sonic-net/sonic-buildimage/pull/17326/

There is also another commit by Cisco #15153  which added a smartswitch-t1 topo recently. I am trying to get them to change it to "t1-smartswitch" to be consistent everywhere.

**Comment by ganglyu** [comment]
Is this topology used exclusively for KVM? For t1, we have various topologies such as t1-4-lag, t1-8-lag, etc. Which 11 topology is supported by this t1-smartswitch?

File: ansible/roles/vm_set/tasks/start_dpu_vm.yml [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+      msg: [
+        "Failed, no working SONiC image download URL is found. There are 2 options to fix it:",
+        "  1. Fix vsonic_image_url defined in ansible/group_vars/vm_host/sonic.yml",
+        "  2. Manually put SONiC image to {{ root_path }}/images/{{ vsonic_image_filename }}",
+      ]
+    when: working_image_urls | length == 0
+
+  - name: Download SONiC image from the first URL in working_image_urls
+    get_url:
+      url: "{{ working_image_urls[0] }}"
```
[the remark this answers is not in the document]
Comment (zjswhhh) [reply ansible/roles/vm_set/tasks/start_dpu_vm.yml:57]: Will address in a different PR. MSFT ADO#: 26584114

**Review by ganglyu — APPROVED** [review]

**Comment by zjswhhh** [comment]
Hi @yxieca - can you help merge please?

**Pull request merged**